test/unit/test_host.rb

```diff
@@ -65,7 +65,6 @@ def test_assert_hosts_hash_equally
     end
 
     def test_assert_hosts_compare_equal
-      assert Host.new('example.com') == Host.new('example.com')
```

Comment:
Maybe the warning is occurring because `==` is [overwritten as an alias](https://github.com/capistrano/sshkit/blob/master/lib/sshkit/host.rb#L74) and rubocop can't detect that. In any case, I think we could do with some more tests, especially testing that different hosts aren't equal. Testing different user, hostname, and port combinations would probably be a good idea - my preference would probably be to add these to `test_assert_hosts_hash_equally` rather than testing 3 times for all 3 equal methods.